src/main/java/net/openhft/chronicle/threads/EventGroup.java

```diff
@@ -307,53 +283,52 @@ public void addTimingMonitor(final String description,
      * (or timing out)
      */
     @Override
-    public synchronized void start() {
-        throwExceptionIfClosed();
-
-        if (!isAlive()) {
-            if (core != null)
-                core.start();
-            if (blocking != null)
-                blocking.start();
+    protected void performStart() {
+        if (core != null)
+            core.start();
+        if (blocking != null)
+            blocking.start();
 
-            if (replication != null)
-                replication.start();
+        if (replication != null)
+            replication.start();
 
-            for (VanillaEventLoop concThread : concThreads) {
-                if (concThread != null)
-                    concThread.start();
-            }
+        for (VanillaEventLoop concThread : concThreads) {
+            if (concThread != null)
+                concThread.start();
+        }
 
-            monitor.start();
-            // this checks that the core threads have stalled
-            if (core != null)
-                addThreadMonitoring(MONITOR_INTERVAL_MS, core);
-
-            // wait for core to start, We use a TimingPauser, previously we waited for ever
-            TimingPauser timeoutPauser = Pauser.sleepy();
-            while (!isAlive()) {
-                try {
-                    timeoutPauser.pause(WAIT_TO_START_MS, TimeUnit.MILLISECONDS);
-                } catch (TimeoutException e) {
-                    throw Jvm.rethrow(e);
-                }
+        monitor.start();
+        // this checks that the core threads have stalled
+        if (core != null)
+            addThreadMonitoring(MONITOR_INTERVAL_MS, core);
+
+        // wait for core to start, We use a TimingPauser, previously we waited for ever
+        TimingPauser timeoutPauser = Pauser.sleepy();
+        while (!isAlive()) {
+            try {
+                timeoutPauser.pause(WAIT_TO_START_MS, TimeUnit.MILLISECONDS);
+            } catch (TimeoutException e) {
+                throw Jvm.rethrow(e);
             }
         }
     }
 
     @Override
-    public void stop() {
+    protected void performStopFromNew() {
+        performStop();
+    }
+
+    @Override
+    protected void performStopFromStarted() {
+        performStop();
+    }
+
+    private void performStop() {
         monitor.stop();
-        if (replication != null)
-            replication.stop();
-        for (VanillaEventLoop concThread : concThreads) {
-            if (concThread != null)
-                concThread.stop();
-        }
-        if (core != null)
-            core.stop();
-        if (blocking != null)
-            blocking.stop();
+        Stream.concat(concThreads.stream(), Stream.of(replication, core, blocking))
+                .filter(Objects::nonNull)
+                .parallel()
+                .forEach(EventLoop::stop);
```

Comment:
No reason we shouldn't close all of these in parallel? I think it's important we stop the monitor thread first though